app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  before_save :downcase_email
+  validates :email, presence: true, length: {minium: 20, maximum: 40},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  validates :name, presence: true, length: {maximum: 10}
+  validates :password, presence: true, length: {minimum: 6}, if: :password
+  has_secure_password
+  private
+  def downcase_email
+    email.downcase!
+  end
```

Comment:
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
  
  before_save :downcase_email
  
  validates :email, presence: true, length: {minium: 20, maximum: 40},
    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
  validates :name, presence: true, length: {maximum: 10}
  validates :password, presence: true, length: {minimum: 6}, if: :password
  
  has_secure_password
  
  private
  
  def downcase_email
    email.downcase!
  end
```
1. như Sếp nói: thì nên "buông đôi tay nhau" ra nhé 

 2. Các giá trị như 40 20 10 ... em mạnh dạn khai báo vào file `settings `để có thể config tập trung và tái sử dụng nếu cần nhé
 Để bjk file config ở đâu thì em tìm hiểu cho chị gem config nè
 ở đây chị có thêm 1 suggest là thay vì em khai báo các biến như `user.email.max_length: 40` thì mình nên khai báo clear giá trị và tránh focus để tái sử dụng như `digits.length_40: 40` ri sẽ hợp lý hơn nhé